File: apps/api/schemas/evidencias.py
from pydantic import BaseModel
from typing import Optional, List

from enum import Enum

# ============= EVIDENCIA SCHEMAS =============

class EstadoEvidencia(str, Enum):
    PENDIENTE = "PENDIENTE"
    APROBADA = "APROBADA"
    RECHAZADA = "RECHAZADA"

class EvidenciaResponse(BaseModel):
    """Evidencia de tarea práctica"""
    id: str
    tareaId: str
    alumnoId: str
    fotoUrl: str
    comentario: Optional[str] = None
    estado: EstadoEvidencia = EstadoEvidencia.PENDIENTE
    feedback: Optional[str] = None
    evaluadorId: Optional[str] = None
    uploadedAt: str
    
    class Config:
        from_attributes = True

class EvaluarEvidenciaRequest(BaseModel):
    """Solicitud para evaluar una evidencia"""
    estado: EstadoEvidencia
    feedback: Optional[str] = None

class UploadEvidenciaResponse(BaseModel):
    """Respuesta al subir evidencia"""
    success: bool
    evidencia: EvidenciaResponse
    message: str

class ListEvidenciasResponse(BaseModel):
    """Lista de evidencias para una tarea"""
    evidencias: List[EvidenciaResponse]
    total: int

File: apps/web/lib/api/evidencias.ts
import { Evidencia, UploadEvidenciaResponse } from '@/types/training';
import { api } from '../api-client';

export const evidenciasApi = {
    /**
     * Subir foto de evidencia
     */
    async uploadEvidencia(
        file: File,
        tareaId: string,
        comentario?: string
    ): Promise<UploadEvidenciaResponse> {
        const formData = new FormData();
        formData.append('file', file);
        formData.append('tareaId', tareaId);
        if (comentario) {
            formData.append('comentario', comentario);
        }

        return api.post('/evidencias/upload', formData);
    },

    /**
     * Confirmar la realización de una tarea que no requiere foto de evidencia
     */
    async confirmarSinFoto(tareaId: string, comentario?: string): Promise<UploadEvidenciaResponse> {
        const formData = new FormData();
        formData.append('tareaId', tareaId);
        if (comentario) {
            formData.append('comentario', comentario);
        }

        return api.post('/evidencias/upload', formData);
    },

    /**
     * Obtener evidencias de una tarea
     */
    async obtenerEvidencias(tareaId: string): Promise<Evidencia[]> {
        const response = await api.get(`/evidencias/tarea/${tareaId}`);
        return response.evidencias;
    },

    /**
     * Eliminar evidencia
     */
    async eliminarEvidencia(id: string): Promise<{ success: boolean; message: string }> {
        return api.delete(`/evidencias/${id}`);
    },

    /**
     * Listar evidencias pendientes de revisión (Instructor)
     */
    async listarRevisiones(): Promise<Evidencia[]> {
        const response = await api.get('/evidencias/revision');
        return response.evidencias;
    },

    /**
     * Evaluar evidencia (Instructor)
     */
    async evaluarEvidencia(
        id: string,
        data: { estado: 'APROBADA' | 'RECHAZADA'; feedback?: string }
    ): Promise<Evidencia> {
        return api.put(`/evidencias/${id}/evaluar`, data);
    },

    /**
     * Estadísticas de revisión para el dashboard del instructor
     */
    async obtenerStats(): Promise<{ pending: number; totalReviewed: number; activeAlumnos: number; cursosAsignados: number; credencialesEmitidas: number }> {
        return api.get('/evidencias/stats');
    },
};
